frontend/pulse3d/src/components/basicWidgets/UsageProgressWidget.js

```diff
@@ -20,52 +20,69 @@ const ExpiredP = styled.p`
 `;
 
 export default function UsageProgressWidget({ colorOfTextLabel }) {
-  const { usageQuota } = useContext(AuthContext);
-  const [maxUploads, setMaxUploads] = useState(-1);
-  const [actualUploads, setActualUploads] = useState(0);
+  const { usageQuota, setUsageQuota } = useContext(AuthContext);
+  const [maxJobs, setMaxUploads] = useState();
+  const [actualJobs, setActualUploads] = useState(0);
   const [usagePercentage, setUsagePercentage] = useState(0);
-
+  const [isExpired, setIsExpired] = useState();
+  const pollUsageQuota = async () => {
+    try {
+      const response = await fetch(`${process.env.NEXT_PUBLIC_PULSE3D_URL}/usage-quota`);
+      if (response && response.status === 200) {
+        const newUsageQuota = await response.json();
+        const limit = parseInt(newUsageQuota.limits.jobs);
+        const actual = parseInt(newUsageQuota.current.jobs);
+        const usagePercentage = Math.round((actual / limit) * 100);
```

Comment:
since usagePercentage is already being used as a variable name in state, I would name this something else so there isn't any unexpected behaviors